src/error.rs

```diff
@@ -24,6 +27,10 @@ pub enum EvalError<'tcx> {
     Unimplemented(String),
     DerefFunctionPointer,
     ExecuteMemory,
+    ArrayIndexOutOfBounds(Span, u64, u64),
+    Math(Span, ConstMathErr),
+    InvalidBitShiftRhs(PrimVal),
+    Overflow(PrimVal, PrimVal, mir::BinOp, PrimVal),
```

Comment:
Since the current tests are only run in debug mode, the overflow checks prevent the overflow error to ever trigger.